internal/auth/oidc/repository_managed_group_members_test.go

```diff
@@ -0,0 +1,261 @@
+package oidc_test
+
+import (
+	"context"
+	"math/rand"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/boundary/internal/auth/oidc"
+	"github.com/hashicorp/boundary/internal/auth/oidc/store"
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/iam"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/servers/controller"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const testFakeManagedGroupFilter = `"/foo" == "bar"`
+
+func Test_ManagedGroupMemberships(t *testing.T) {
+	// This tests both managed group membership functions (set/list) as list is
+	// always called as a return from set and we are validating the values that
+	// come back against what we expect.
+
+	// This test can be run in parallel; the subtests *cannot*.
+	t.Parallel()
+
+	// Note: using a test controller here for ease of setup as we need a working
+	// dev OIDC auth method and associated accounts. This test is not making API
+	// calls! It's accessing the repo directly via the test controller's
+	// exposure of the underlying DB primitives.
+	tc := controller.NewTestController(t, nil)
+	defer tc.Shutdown()
+
+	conn := tc.DbConn()
+	rw := db.New(conn)
+
+	kmsCache := tc.Kms()
+	iamRepo := tc.IamRepo()
+	org, _ := iam.TestScopes(t, iamRepo)
+
+	ctx := context.Background()
+	databaseWrapper, err := kmsCache.GetWrapper(ctx, org.PublicId, kms.KeyPurposeDatabase)
+	require.NoError(t, err)
+
+	authMethod := oidc.TestAuthMethod(
+		t, conn, databaseWrapper, org.GetPublicId(), oidc.ActivePrivateState,
+		"alice-rp", "fido",
+		oidc.WithSigningAlgs(oidc.RS256),
+		oidc.WithIssuer(oidc.TestConvertToUrls(t, "https://www.alice.com")[0]),
+		oidc.WithApiUrl(oidc.TestConvertToUrls(t, "https://www.alice.com/callback")[0]),
+	)
+
+	repo, err := oidc.NewRepository(rw, rw, kmsCache)
+	require.NoError(t, err)
+	require.NotNil(t, repo)
+
+	mgs := make([]*oidc.ManagedGroup, 0, 10)
+
+	for i := 0; i < 100; i++ {
+		mg := oidc.AllocManagedGroup()
+		mg.AuthMethodId = authMethod.PublicId
+		mg.Filter = testFakeManagedGroupFilter
+		got, err := repo.CreateManagedGroup(context.Background(), org.GetPublicId(), mg)
+		require.NoError(t, err)
+		mgs = append(mgs, got)
+	}
+
+	// Fetch valid OIDC accounts. One will be "static" where we will simply
+	// ensure modifying the groups for the other doesn't affect it; the other
+	// will be used for testing.
+	rows, err := rw.Query(ctx, "select public_id from auth_oidc_account limit 2", nil)
```

Comment:
you could refactor this to just use vanilla db.SearchWhere(...) WithLimit(2) and an []Account